### PR TITLE
Done with basic setup to integrate map to our website

### LINKED ISSUE #10: Add a map showing all the countries that fellows are located in
Underneath projects and fellows, have an interactive map that displays where everyone is located.

- [ ] When you click on the pin location for that fellow, it gives you a card with information about the fellow (similar to the fellow section

You can chose to have pins or use circular profile pictures to show where everyone is on the map. You could remove the fellows section and replace it with the map.

### PR DESCRIPTION
**Pull request by Sargam-Agarwal**
Adds the map to the website 
Fixes #10 

## Please, go through these steps before you submit a PR.

- [x] My Pod Leader knows I'm working on this Pull Request
- [x] I've explained what the Pull Request is adding.
- [x] I've explained why this is important.

### CODE REVIEW

File: _includes/header.html
Comment (NoahCardoza) [inline _includes/header.html:22]: Some browsers may take issue with this src tag. They will throw "Mixed Content" errors if you try to access an HTTP resource from an HTTPS website. It's generally best to use HTTPS over HTTP wherever possible.

For example, here are Chrome's logs:

image

File: _includes/header.html
Comment (NoahCardoza) [inline _includes/header.html:34]: Same here.

**Review by NoahCardoza — CHANGES_REQUESTED** [review]
Make sure to be careful mixing HTTP and HTTPS content. It might work locally since you are accessing the site over HTTP but it won't work on production over HTTPS.

File: assets/js/map.js
Comment (NoahCardoza) [inline assets/js/map.js:11]: Looks like another `http://` slipped in here.

It is causing these warnings whenever you look around on the map: 
image

**Review by NoahCardoza — CHANGES_REQUESTED** [review]

File: assets/js/map.js
Comment (Sargam-Agarwal) [reply assets/js/map.js:11]: Ahh okay. I'll see to it.

File: assets/js/map.js
Comment (NoahCardoza) [inline assets/js/map.js:16]: While this works, I think it would be best to store this information in the data files.

**Review by NoahCardoza — CHANGES_REQUESTED** [review]
Before proceeding, I'd wait until #25 is merged so we don't cause conflicts when editing the data files. I'll let you know when it's safe to proceed. Until then feel free to take up another issue!

Everything else looks good here!

**Comment by NoahCardoza** [comment]
I nearly forgot! 

Please also reset the Gemfile.lock file. Since this PR doesn't add any new dependencies or revolve around fixing the build issues, we shouldn't be pushing it. We should open a separate PR for that.

File: assets/js/map.js
Comment (Sargam-Agarwal) [reply assets/js/map.js:16]: Umm, then may be when #25 will be merged, then all the rqd data will automatically be there that ill need. So i wont need to add anything else. So, ill fetch the data from there only once I pull all the changes on my local after issue 25 is merged.

**Comment by Sargam-Agarwal** [comment]
@NoahCardoza reset means deleting the file ? That is what my partner and I could infer out of it.

**Comment by NoahCardoza** [comment]
Close! I'm not all that familiar with Ruby (compared to other languages) but generally .lock files are OS agnostic. What is going on here is that the dependencies were out of date, so most fellows needed to update them. That being said, to handle it properly a new issue should be made that addresses that issue. We shouldn't be updating dependencies when we add a new feature. Especially in this case, since some fellows might not push the update file (and vice versa) it could cause some nasty merge conflicts on that file.

File: assets/js/map.js
Comment (NoahCardoza) [reply assets/js/map.js:16]: Alright, #25 was just merged so you should be good to finish this up!

Good luck!

**Comment by NoahCardoza** [comment]
It might also be cool to show the fellow's profile picture in the card that shows up when clicking the map marker.

**Comment by Sargam-Agarwal** [comment]
Umm yess! I was thinking of customizing the marker itself with the profile pictures. What sounds better !?

**Comment by NoahCardoza** [comment]
Ohoo! I like your idea better!

Can't wait to see how it turns out!

**Comment by Sargam-Agarwal** [comment]
Umm @NoahCardoza, I am not able to figure out the way to fetch the data from yaml file into js file. I tried to go through the pod portfolios but most of them have done it by explicit definition of the data in the map.js file.

**Comment by Pratyush1606** [comment]
Hey @Sargam-Agarwal, you can use something like this:
```
<div>
    {% for item in site.data.fellows %}
    <script>
        if("{{ item.name }}"==fellowName){
            name = "{{ item.name }}";
            img = "{{ item.img }}";
            loc = "{{ item.location }}";
            description = "{{ item.description }}";
            fav_programming_language = "{{ item.fav_programming_language }}";
            interests_outside_tech = "{{ item.interests_outside_tech }}";
            university =  "{{ item.university }}";
            linkedin = "{{ item.linkedin }}";
            twitter = "{{ item.twitter }}";
            github = "{{ item.github }}";
        }
    </script>
    {% endfor %}
</div>
```
We implemented this method in [fellow.html](https://github.com/Pratyush1606/pod-4.1.4-portfolio/blob/020f40dda45c00545f191f1acfe31510d67080d3/fellow.html) to display a separate page for every fellow.

**Comment by Sargam-Agarwal** [comment]
Hey @Pratyush1606 
Thank you for this. I tried this out but this seems to work only for html files not in js files :(
Its giving errors with the syntax when used in js.

File: index.html
Comment (NoahCardoza) [inline index.html:13]: Looks like when you resolved the merge conflicts it duplicated this line.

File: index.html
Comment (NoahCardoza) [inline index.html:15]: Same here.

**Review by NoahCardoza — CHANGES_REQUESTED** [review]
If we were to include these we'd have to of the same sections on the landing page.

**Comment by NoahCardoza** [comment]
> Hey @Pratyush1606 Thank you for this. I tried this out but this seems to work only for html files not in js files :( Its giving errors with the syntax when used in js.

You could try to put:

```html
<script>
  const DATA = {{ site.data | jsonify }};
</script>
```

In side `/index.html` and then in your `map.js` access the `DATA` object which will include all the fellow data.

**Comment by Sargam-Agarwal** [comment]
@NoahCardoza 
I have made the required changes and the map seems to work just as expected. Kindly see to it now. Hope now its ready to be merged 🥲

File: assets/js/map.js
Comment (NoahCardoza) [inline assets/js/map.js:18]: Generally, it's not a good practice to leave debug statements in production code.

File: assets/js/map.js
Comment (NoahCardoza) [inline assets/js/map.js:16]: And another one of those pesky comments!

**Review by NoahCardoza — CHANGES_REQUESTED** [review]
Just wanted to point these out to you but I'll make the edits myself since they are tiny. Great world over all!

**Comment by NoahCardoza** [comment]
One last thing, do you know if there is a way to fix the aspect ratio on these images? I'm not very familiar with Open Street Map but it would be nice if we could clean that up a little.

If not, no problem. I can merge it as is.

Here is an example of what I'm talking about:

image

image

File: assets/js/map.js
Comment (Sargam-Agarwal) [reply assets/js/map.js]: Ah yess! will take care of it from next time onwards 👍🏻

File: assets/js/map.js
Comment (Sargam-Agarwal) [reply assets/js/map.js]: Yup 😅

**Comment by Sargam-Agarwal** [comment]
Sure @NoahCardoza.
I'll see to it 👍🏻

**Comment by Sargam-Agarwal** [comment]
@NoahCardoza 
There doesn't seem to be any provision to improve the icon quality by Leaflet API. 
[https://leafletjs.com/reference.html#icon](https://leafletjs.com/reference.html#icon) - these are all the options available to us for customizing our marker. I tried to play around with the width and height values for the icon and also the border radius values but that only changes the shape of the icon, not the quality of the image.

**Comment by NoahCardoza** [comment]
The only thing I can think of is cropping all the images to the same aspect ratio... but that's something for another issue/PR I think.

**Pull request merged**
